src/parse/asp/interpreter.go

```diff
@@ -236,9 +245,14 @@ func (i *interpreter) optimiseExpressions(stmts []*Statement) {
 			if expr.Val.Property == nil && len(expr.Val.Ident.Action) == 0 {
 				expr.Optimised = &OptimisedExpression{Local: expr.Val.Ident.Name}
 				return false
-			} else if expr.Val.Ident.Name == "CONFIG" && len(expr.Val.Ident.Action) == 1 && expr.Val.Ident.Action[0].Property != nil && len(expr.Val.Ident.Action[0].Property.Action) == 0 {
-				expr.Optimised = &OptimisedExpression{Config: expr.Val.Ident.Action[0].Property.Name}
-				expr.Val = nil
+			} else if expr.Val.Ident.Name == "CONFIG" && len(expr.Val.Ident.Action) == 1 && expr.Val.Ident.Action[0].Property != nil {
+				if prop := expr.Val.Ident.Action[0].Property; len(prop.Action) == 0 {
+					expr.Optimised = &OptimisedExpression{Config: prop.Name}
+					expr.Val = nil
+				} else if len(prop.Action) == 1 && prop.Action[0].Property != nil && len(prop.Action[0].Property.Action) == 0 {
+					expr.Optimised = &OptimisedExpression{Config: prop.Name, SubConfig: prop.Action[0].Property.Name}
+					expr.Val = nil
+				}
```

Comment:
Nice! Does this make a big difference? I imagine we do this quite a lot of so it would help.

Reply:
I think a bit of one. We're probably approaching a point where there are more three-level config lookups than two so seemed useful.